rp2040-hal/src/dma/single_channel.rs

```diff
@@ -31,6 +31,14 @@ pub trait SingleChannel: Sealed {
         }
     }
 
+    /// Check if the DMA_IRQ_0 signal for this channel is enabled.
+    fn is_enabled_irq0(&mut self) -> bool {
+        // Safety: We only use the atomic alias of the register.
+        unsafe {
+            (*DMA::ptr()).inte0().read().bits() & 1 << self.id() != 0
+        }
```

Comment:
I'd feel more comfortable if this wasn't relying on operator precedence and order of operations.
```suggestion
        let enabled = unsafe {(*DMA::ptr()).inte0().read().bits() };
        (enabled & (1 << self.id())) != 0
```

Reply:
Done! Thank you for catching this